packages/gatsby/cache-dir/component-renderer.js

```diff
@@ -117,6 +118,24 @@ class ComponentRenderer extends React.Component {
       return true
     }
 
+    if (Object.keys(this.props).length !== Object.keys(nextProps).length) {
```

Comment:
minor thing but it couldn't hurt to assign to a const here - e.g.

    const propsKeyCount = Object.keys(this.props).length;
    
    if (propsKeyCount !== Object.keys(nextProps).length) { ... }